### PR TITLE
feat: scope

### LINKED ISSUE #978: rune cross-execution deduplication

### PR DESCRIPTION
**Pull request by tjjfvi**
Resolves #978

### CODE REVIEW

**Comment by harrysolovay** [comment]
Should we provide access to scope within the Rune system? (to enable the fluent API and patterns to decide how/when to cache-invalidate)

```ts
import { $accountInfo, polkadotDev } from "@capi/polkadot-dev"
import { $, createDevUsers, Scope } from "capi"

const { alexa } = await createDevUsers()

const scope = new Scope()

const accountInfo = polkadotDev.System.Account.value(alexa.publicKey)

const a = await accountInfo.run(scope)
await delay(30000)
const b = await accountInfo.run(scope) // could be different now
await delay(30000)
const c = await accountInfo.run(scope) // could be different yet again
```


How might one specify whether to rerun all occurrences, vs. idempotent per run, vs. per scope?

**Comment by tjjfvi** [comment]
For now, these kinds of questions of out of scope (pun intended) until the rest of the caching/memoization system is more settled.

File: Readme.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ Retrieve the first 10 entries from a storage map of Polkadot.
 ```ts
 import { polkadot } from "@capi/polkadot"
```
Comment (harrysolovay) [inline Readme.md:99]:
```suggestion
import { polkadot } from "@capi/polkadot"
import { Scope } from "capi"
```

File: Readme.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ Retrieve the first 10 entries from a storage map of Polkadot.
...
 
-const accounts = await polkadot.System.Account.entries({ limit: 10 }).run()
+const accounts = await polkadot.System.Account.entries({ limit: 10 }).run(scope)
```
Comment (harrysolovay) [inline Readme.md:101]:
```suggestion
const accounts = await polkadot.System.Account.entries({ limit: 10 }).run(new Scope())
```

**Review by harrysolovay — APPROVED** [review]

File: Readme.md [hunk 3]
```diff
@@ -97,8 +97,9 @@ Retrieve the first 10 entries from a storage map of Polkadot.
 
 ```ts
 import { polkadot } from "@capi/polkadot"
+import { Scope } from "capi"
 
-const accounts = await polkadot.System.Account.entries({ limit: 10 }).run()
+const accounts = await polkadot.System.Account.entries({ limit: 10 }).run(new Scope())
```
Comment (harrysolovay) [inline Readme.md:102]:
```suggestion
const accounts = await polkadot.System.Account
  .entries({ limit: 10 })
  .run(new Scope())
```

**Review by harrysolovay — APPROVED** [review]

**Review by harrysolovay — APPROVED** [review]

**Review by kratico — APPROVED** [review]
LGTM

**Pull request merged**